hardware/ni_x_series_gated_counter.py

```diff
@@ -0,0 +1,444 @@
+# -*- coding: utf-8 -*-
+
+"""
+This file contains the Qudi Hardware module NICard class.
+
+Qudi is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Qudi. If not, see <http://www.gnu.org/licenses/>.
+
+Copyright (c) the Qudi Developers. See the COPYRIGHT.txt file at the
+top-level directory of this distribution and at <https://github.com/Ulm-IQO/qudi/>
+"""
+
+import numpy as np
+import re
+
+import PyDAQmx as daq
+
+from core.module import Base, ConfigOption
+from interface.slow_counter_interface import SlowCounterInterface
+from interface.slow_counter_interface import SlowCounterConstraints
+from interface.slow_counter_interface import CountingMode
+
+
+class NationalInstrumentsXSeriesGatedCounter(Base, SlowCounterInterface):
+    """ A National Instruments device that can count and control microvave generators.
+
+    !!!!!! NI USB 63XX, NI PCIe 63XX and NI PXIe 63XX DEVICES ONLY !!!!!!
+
+    See [National Instruments X Series Documentation](@ref nidaq-x-series) for details.
+
+    Example config for copy-paste:
+
+    nicard_6343:
+        module.Class: 'ni_x_series_counter.NationalInstrumentsXSeriesGatedCounter'
+        photon_sources: '/Dev1/PFI8'
+        counter_channel: '/Dev1/Ctr1'
+        gate_in_channel: '/Dev1/PFI7'
+
+        default_samples_number: 50
+        max_counts: 3e7
+        read_write_timeout: 10
+        counting_edge_rising: True
+
+    """
+
+    _modtype = 'NICard'
+    _modclass = 'hardware'
+
+    # config options
+    _photon_source = ConfigOption('photon_sources', missing='error')
+
+    _counter_channel = ConfigOption('counter_channels', missing='error')
+
+    _gate_in_channel = ConfigOption('gate_in_channel', missing='error')
+    # number of readout samples, mainly used for gated counter
+    _default_samples_number = ConfigOption('default_samples_number', 50, missing='info')
+
+    _max_counts = ConfigOption('max_counts', default=3e7)
+    # timeout for the Read or/and write process in s
+    _RWTimeout = ConfigOption('read_write_timeout', default=10)
+    _counting_edge_rising = ConfigOption('counting_edge_rising', default=True)
+
+    def on_activate(self):
+        """ Starts up the NI Card at activation.
+        """
+        # the tasks used on that hardware device:
+        self._gated_counter_daq_task = None
+
+    def on_deactivate(self):
+        """ Shut down the NI card.
+        """
+        self.close_counter()
+        self.close_clock()
```

Comment:
nope, still no hardware reset.

Even if the close_clock currently has a pass, it should still be called, as it might get other functionality later on.